src/sonic-config-engine/portconfig.py

```diff
@@ -423,9 +426,10 @@ def parse_platform_json_file(hwsku_json_file, platform_json_file):
 
         # take optional fields from hwsku.json
         for child_port in child_ports:
-            for key, item in hwsku_dict[INTF_KEY][child_port].items():
-                if key in OPTIONAL_HWSKU_ATTRIBUTES:
-                    child_ports.get(child_port)[key] = item
```

Comment:
Consider this case, Ethernet0 -> {Ethernet0 & Ethernet4}. Default computation would infer the value to be '1' for Ethernet0 but Ethernet0 according to the hwsku.json would be "0". So, i think letting the value of hwsku.json overriding the computed value wouldn't work.

Reply:
Are you suggesting I remove hwsku.json optional subport value altogether ?

Reply:
Yes, if we can infer subport by looking at lanes directly i don't see a need for hwsku.json subport value.

Reply:
Removed subport as an optional value